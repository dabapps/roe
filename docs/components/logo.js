var React = require('react');
var Styled = require('rsg-components/Styled').default;

function styles (style) {
  return {
    image: {
      fontFamily: style.fontFamily.base,
      width: 63,
      height: 24,
      display: 'block'
    }
  };
}

function Logo (props) {
  return React.createElement(
    'a',
    {
      href: '#'
    },
    React.createElement(
      'img',
      {
        src: 'images/roe-logo-small.png',
        alt: 'Roe Logo',
        className: props.classes.image
      }
    )
  );
}

module.exports = Styled(styles)(Logo);
