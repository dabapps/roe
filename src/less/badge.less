.badge-overrides() {
  min-width: 1.7em;
  text-align: center;
  cursor: default!important;
}

.create-badge-pew {
  display: inline-block;
  height: auto;
  padding: 5px 10px;
  font-family: arial,helvetica,sans-serif;
  font-size: 1em;
  line-height: 1.3em;
  font-weight: normal;
  text-decoration: none;
  border-radius: @badge-border-radius;
}

.create-badge-styles (
  @background: @badge-background-default,
  @color: @badge-text-color-dark,
) {
  .create-badge-pew();
  background-color: @background;
  color: @color;
}

.create-badge (@type, @background, @color) {
  &.@{type} {
    .create-badge-styles(@background, @color);
  }
}

.badge {
  .create-badge-styles();
  .create-badge(primary, @primary, @badge-text-color-light);
  .create-badge(secondary, @secondary, @badge-text-color-light);
  .create-badge(tertiary, @tertiary, @badge-text-color-light);
  .create-badge(info, @info, @badge-text-color-light);
  .create-badge(success, @success, @badge-text-color-light);
  .create-badge(warning, @warning, @badge-text-color-light);
  .create-badge(error, @error, @badge-text-color-light);
  .badge-overrides();

  &.large {
    font-size: @font-size-large;
    padding: @padding-large / 2 @padding-large;
  }

  &.small {
    font-size: @font-size-small;
    padding: @padding-small / 2 @padding-small;
  }
}
