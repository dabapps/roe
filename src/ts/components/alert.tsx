import * as classNames from 'classnames';
import * as React from 'react';
import { HTMLProps, StatelessComponent } from 'react';
import { ComponentProps } from '../types';

export type AlertProps = ComponentProps & HTMLProps<HTMLElement>;

/**
 * A component for applying various styles to text, ideal for info, success, and error messages.
 */
export const Alert: StatelessComponent<AlertProps> = (props) => {
  const { children, className, component: Component = 'div', ...remainingProps } = props;

  return (
    <Component {...remainingProps} className={classNames(['alert', className])}>
      {children}
    </Component>
  );
}

export default Alert;
