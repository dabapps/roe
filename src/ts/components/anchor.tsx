import * as React from 'react';
import { HTMLProps, StatelessComponent } from 'react';
import { getHref } from '../utils';

export type AnchorProps = HTMLProps<HTMLElement>;

export const Anchor: StatelessComponent<AnchorProps> = (props) => {
  const { children, href, ...remainingProps } = props;

  const automaticHref = getHref(children, href);

  return (
    <a {...remainingProps} id={automaticHref} href={automaticHref ? '#' + automaticHref : undefined}>
      {children}
    </a>
  );
}
