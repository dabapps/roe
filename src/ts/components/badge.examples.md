#### Example

```js
<div>
  <p>
    <SpacedGroup block>
      <Badge>
        1.00
      </Badge>
      <Badge>
        Default
      </Badge>
      <Badge className="primary">
        Primary
      </Badge>
      <Badge className="secondary">
        Secondary
      </Badge>
      <Badge className="tertiary">
        Tertiary
      </Badge>
      <Badge className="info">
        Info
      </Badge>
      <Badge className="success">
        Success
      </Badge>
      <Badge className="warning">
        Warning
      </Badge>
      <Badge className="error">
        Error
      </Badge>
    </SpacedGroup>
  </p>
</div>
```

#### Less variables

```less
@badge-border-radius: @font-size-base * 2;
@badge-background-default: @grey-lighter;
@badge-text-color-light: @grey-lightest;
@badge-text-color-dark: @grey-dark;
```
