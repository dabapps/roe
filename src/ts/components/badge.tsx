import * as classNames from 'classnames';
import * as React from 'react';
import { HTMLProps, PureComponent } from 'react';
import { ComponentProps } from '../types';

export type BadgeProps = ComponentProps & HTMLProps<HTMLElement>;
/**
 * Badge Component
 */
export class Badge extends PureComponent<BadgeProps, {}> {
  public render() {
    const {
      className,
      children,
      component: Component = 'span',
      ...remainingProps
    } = this.props;

    return (
      <Component {...remainingProps} className={classNames('badge', className)}>
        {children}
      </Component>
    );
  }
}

export default Badge;
