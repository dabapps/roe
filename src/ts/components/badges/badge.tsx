import * as classNames from 'classnames';
import * as React from 'react';
import { HTMLProps, PureComponent } from 'react';
import { ComponentProps } from '../../types';

export interface BadgeProps extends ComponentProps, HTMLProps<HTMLElement> {
  /**
   * Make the Badge small
   */
  small?: boolean;
  /**
   * Make the Badge large
   */
  large?: boolean;
}

/**
 * Badge Component
 */
export class Badge extends PureComponent<BadgeProps, {}> {
  public render() {
    const {
      className,
      small,
      large,
      children,
      component: Component = 'div',
      ...remainingProps
    } = this.props;

    return (
      <Component
        {...remainingProps}
        className={classNames(
          'badge',
          small ? 'small' : null,
          large ? 'large' : null,
          className
        )}
      >
        {children}
      </Component>
    );
  }
}

export default Badge;
