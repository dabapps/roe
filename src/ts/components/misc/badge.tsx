import * as classNames from 'classnames';
import * as React from 'react';
import { HTMLProps, PureComponent } from 'react';
import { ComponentProps } from '../../types';

export type BadgeProps = ComponentProps & HTMLProps<HTMLElement>;
/**
 * A badge component for displaying small pieces of information such as counts and statuses.
 */
export class Badge extends PureComponent<BadgeProps, {}> {
  public render() {
    const {
      className,
      children,
      component: Component = 'span',
      ...remainingProps
    } = this.props;

    return (
      <Component {...remainingProps} className={classNames('badge', className)}>
        {children}
      </Component>
    );
  }
}

export default Badge;
