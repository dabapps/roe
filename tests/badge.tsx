import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Badge } from '../src/ts/components/badge';

describe('Badge', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Badge>Default</Badge>);

    expect(tree).toMatchSnapshot();
  });

  it('should take regular element attributes', () => {
    const tree = renderer.create(<Badge className="primary">Primary</Badge>);

    expect(tree).toMatchSnapshot();
  });
});
