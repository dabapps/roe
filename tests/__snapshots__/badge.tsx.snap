// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should match snapshot 1`] = `
<span
  className="badge"
>
  Default
</span>
`;

exports[`Badge should take regular element attributes 1`] = `
<span
  className="badge primary"
>
  Primary
</span>
`;
